### PR TITLE
Add RBAC to cluster and manager modules 

### PR DESCRIPTION
**Pull request by davidjiglesias**
|Related issue|
|---|
|#4126|

## Description

Hello team, this PR adds RBAC refactor to cluster and manager endpoints and modules.

## Tests

Tests are handled in #4127(Cluster) and #4129(Manager)

### CODE REVIEW

File: api/api/controllers/cluster_controller.py [hunk 1]
```diff
@@ -322,11 +268,7 @@ def get_stats_node(node_id, pretty=False, wait_for_complete=False, date=None):
     :param date: Selects the date for getting the statistical information. Format YYYY-MM-DD.
     """
     if date:
-        try:
-            year, month, day = date.split('-')
-        except ValueError:
-            raise WazuhError(1412)
-
+        year, month, day = date.split('-')
```
Comment (crd1985) [inline api/api/controllers/cluster_controller.py]: We should better transform the date to a real `date` object and then extract year, month and day from it via properties.

File: framework/wazuh/core/cluster/dapi/dapi.py [hunk 2]
```diff
@@ -45,17 +49,18 @@ def __init__(self, f: Callable, logger: logging.getLogger, f_kwargs: Dict = None
...
         self.f_kwargs = f_kwargs if f_kwargs is not None else {}
         self.node = node if node is not None else local_client
-        self.cluster_items = cluster.get_cluster_items() if node is None else node.cluster_items
+        self.cluster_items = wazuh.core.cluster.utils.get_cluster_items() if node is None else node.cluster_items
         self.debug = debug
         self.pretty = pretty
-        self.node_info = cluster.get_node() if node is None else node.get_node()
-        self.request_id = str(random.randint(0, 2**10 - 1))
+        self.node_info = wazuh.core.cluster.cluster.get_node() if node is None else node.get_node()
+        self.request_id = str(random.randint(0, 2 ** 10 - 1))
```
Comment (crd1985) [inline framework/wazuh/core/cluster/dapi/dapi.py]:
```suggestion
        self.request_id = str(random.randint(0, 2**10 - 1))
```

File: framework/wazuh/core/cluster/dapi/dapi.py [hunk 3]
```diff
@@ -470,3 +492,16 @@ def add_request(self, request: bytes):
...
+    try:
+        lc = local_client.LocalClient()
+        try:
+            result = await control.get_nodes(lc)
+        finally:
+            lc.transport.close()
+
+        return [node['name'] for node in result['items']]
+    except Exception:
+        raise WazuhInternalError(3012)
```
Comment (crd1985) [inline framework/wazuh/core/cluster/dapi/dapi.py]: I would move this function to `control.py`

File: framework/wazuh/rbac/decorators.py [hunk 4]
```diff
@@ -235,6 +238,7 @@ def _get_required_permissions(actions: list = None, resources: list = None, **kw
                 target_params[m.group(1)] = m.group(2)
             else:  # Static resource
                 target_params[m.group(1)] = '*'
+                add_denied = not broadcast.get()
```
Comment (crd1985) [inline framework/wazuh/rbac/decorators.py]: This line should moved outside the if-else block.

File: framework/wazuh/rbac/decorators.py [hunk 5]
```diff
@@ -320,17 +332,20 @@ def wrapper(*args, **kwargs):
...
                         raise Exception
-                    if target_param != '*':  # No resourceless
+                    if target_param != '*':  # No resourceless and not static
                         kwargs[target_param] = list(allow[res_id])
                     elif len(allow[res_id]) == 0:
                         raise Exception
                 except Exception:
                     if add_denied:
-                        denied = _get_denied(original_kwargs, allow, target_param, res_id)
+                        denied = _get_denied(original_kwargs, allow, target_param, res_id, resources)
```
Comment (crd1985) [inline framework/wazuh/rbac/decorators.py]:
```suggestion
                        denied = _get_denied(original_kwargs, allow, target_param, res_id, resources=resources)
```

File: api/api/controllers/cluster_controller.py (hunk 1 above)
Comment (davidjiglesias) [reply api/api/controllers/cluster_controller.py]: Done

File: framework/wazuh/core/cluster/dapi/dapi.py (hunk 3 above)
Comment (davidjiglesias) [reply framework/wazuh/core/cluster/dapi/dapi.py]: Done

File: framework/wazuh/rbac/decorators.py (hunk 4 above)
Comment (davidjiglesias) [reply framework/wazuh/rbac/decorators.py]: Done

**Pull request merged**